hcache.c

```diff
@@ -799,7 +793,7 @@ mutt_hcache_open(const char *path, const char *folder, hcache_namer_t namer)
   else
   {
     /* remove a possibly incompatible version */
-    if (!stat (path, &sb) && !unlink (path))
+    if (unlink (path) == 0)
     {
       h->ctx = ops->open(path);
```

Comment:
TOCTOU - Time of check vs Time of use.
Don't `stat()` the file, just `unlink()` it.